eventlet/hubs/asyncio.py

```diff
@@ -30,6 +39,18 @@ def __init__(self):
         asyncio.set_event_loop(self.loop)
         self.sleep_event = asyncio.Event()
 
+    def _remove_bad_fds(self):
+        """ Iterate through fds, removing the ones that are bad per the
+        operating system.
+        """
+        all_fds = list(self.listeners[self.READ]) + list(self.listeners[self.WRITE])
+        for fd in all_fds:
+            try:
+                select.select([fd], [], [], 0)
```

Comment:
Using `select()` is problematic. It has a fixed max numeric value for file descriptors. So if the max is e.g. 512 and you pass in file descriptor 637, it won't be able to do anything with that descriptor.

To avoid this you want to use `poll()` if available, and only fall back to `select()` if `poll()` is not available (which effectively means Windows).

Reply:
oh interesting, I was not aware of that threshold. Indeed poll seems more appropriate.
To be honest, I copy/pasted this logic from the other hubs and I tried to adapt it for asyncio. That mean that all the other hubs are impacted by the limitation of `select`.